tools/ci/ci_doxygen.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+import sys
+import filecmp
+import junitparser as junit
+
+# CSDK Doxygen Templates path
+DOXYGEN_TEMPLATES_PATH = "doc/doxygen_templates"
+DOXYGEN_TEMPLATES_FILES = ["style.css", "layout.xml"]
+
+
+def run_cmd(cmd):
+    """
+    Execute the input command on the shell.
+    """
+    print(f"Executing command: {cmd}")
+    try:
+        result = subprocess.run(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            shell=True,
+            encoding="utf-8",
+            check=True,
+            timeout=180,
+        )
+        return result.stdout
+    except subprocess.CalledProcessError as e:
+        result = e.stdout
+        return result
+
+
+def to_junit(results, platform=""):
+    """
+    Convert result to junit format.
+    """
+    report = junit.JUnitXml()
+    if not results:
+        return report
+
+    platform += platform and "."
+
+    for target, target_result in results.items():
+        suite = junit.TestSuite(platform + target)
+        for case_name, case_result in target_result.items():
+            case = junit.TestCase(case_name)
+            if case_result["status"] == "FAIL":
+                case.result = junit.Failure(case_result.get("details", ""))
+            if case_result["status"] == "IGNORE":
+                case.result = junit.Skipped(case_result.get("details", ""))
+            suite.add_testcase(case)
+        report.add_testsuite(suite)
+    report.update_statistics()
+
+    return report
+
+
+def write_junit(junit_report, file_name):
+    try:
+        junit_report.write(file_name, pretty=True)
+    except Exception as err:
+        print(f"[ERROR] Unable to write junit_file: {str(err)}")
+
+
+def check_doxygen_warnings(results, abs_libs_path):
+    """
+    Prints the Doxygen warnings found from running the configuration file in the
+    library under the abs_libs_path. Returns the number of warnings if there are
+    any.
+    """
+    lib_dirs = os.listdir(abs_libs_path)
+    for lib_dir in lib_dirs:
+        abs_lib_path = os.path.join(abs_libs_path, lib_dir)
+        if os.path.exists(os.path.join(abs_lib_path, "doc", "doxygen", "config.doxyfile")):
+            target_result = results.setdefault(lib_dir, {})
```

Comment:
use elaborate key for dict, if multiple parent dir has same lib_dir in them then we will overwrite them

Reply:
They should not, we expect them to be unique